### DISCUSSION TITLE
About Android and iOS Support Source Code for Mesh Agent

### DISCUSSION THREAD

**Discussion by amoljagdalepucsd** [Ylianst/MeshCentral#6371] [Q&A]
Hi @krayon007 ,
                        In this discussion ([https://github.com/Ylianst/MeshCentral/issues/110]), you mentioned that we previously had Android support but had to remove it for several reasons. You also mentioned trying to maintain the control channel in the background for iOS by using VoIP. Were you able to manage the source code for these? If so, could you share it with us so we can take a look.
I need to check it for iOS. @si458 @Ylianst

**Comment by si458** [comment]
The is an Android app which has preview support and very recently the is now background/no consent working but no control support still sadly

As for ios, I've never seen/heard of anyone making/releasing an app sadly

**Reply by amoljagdalepucsd** [reply]
Thanks for the response.
In the discussion [https://github.com/Ylianst/MeshCentral/issues/110](https://github.com/Ylianst/MeshCentral/issues/110) Kyron mentioned that they had issues with maintaining the control channel for iOS in the background, so I asked about it

**Reply by si458** [reply]
I have no idea again about the ios version, you would have to speak to the other person about this.

As for the android version, again yes u can see ur desktop and now see it without consent (only when the user first clicks accept when the device starts up)

But the is still no control support because Google doesn't allow ADB control in its app store, so we have to find another way, the is control access via Android apis, I just haven't worked anything out yet as to what the code is etc 😞

**Reply by silversword411** [reply]
I think iOS and android consider everything not MDM a kludge and on the chopping block for a slow death...so what you really want is a MDM that doesn't have a $3+/month/endpoint cost associated with it. 

Let me know if you find one.
